File: src/main/java/me/lovesasuna/lanzou/network/Downloadable.java
package me.lovesasuna.lanzou.network;

import java.io.File;
import java.net.URL;

/**
 * @author LovesAsuna
 * @date 2020/8/22 11:35
 **/
public interface Downloadable {
    /**
     * 通过链接下载文件
     *
     * @param url 下载复制
     * @return 下载成功与否
     **/
    boolean download(URL url, File file);
}

File: src/main/kotlin/me/lovesasuna/lanzou/network/Downloadable.kt
package me.lovesasuna.lanzou.network

import java.nio.file.Path

/**
 * @author LovesAsuna
 * @date 2020/8/22 11:35
 */
interface Downloadable {
    /**
     * 下载文件
     *
     * @param path
     * @return 下载成功与否
     */
    fun download(path: Path): Boolean
}